plugins/fzuse.rb

```diff
@@ -0,0 +1,150 @@
+require 'socket'
+
+# this is the main routine that's executed in the grandchild process (msfconsole -> fzf -> this)
+if $PROGRAM_NAME == __FILE__
+  exit 64 unless ARGV.length == 2
+
+  UNIXSocket.open(ARGV[0]) do |sock|
+    sock.write ARGV[1] + "\n"
+    sock.flush
+
+    puts sock.read
+  end
+  exit 0
+end
+
+module Msf
+  class Plugin::FuzzyUse < Msf::Plugin
+    class ConsoleCommandDispatcher
+      include Msf::Ui::Console::CommandDispatcher
+
+      FZF_THEME = {
+        'fg' => '-1',
+        'fg+' => 'white:regular:bold',
+        'bg' => '-1',
+        'bg+' => '-1',
+        'hl' => '-1',
+        'hl+' => 'red:regular:bold',
+        'info' => '-1',
+        'marker' => '-1',
+        'prompt' => '-1',
+        'spinner' => '-1',
+        'pointer' => 'blue:bold',
+        'header' => '-1',
+        'border' => '-1',
+        'label' => '-1',
+        'query' => '-1'
+      }.freeze
+
+      def initialize(driver)
+        super
+
+        @module_dispatcher = Msf::Ui::Console::CommandDispatcher::Modules.new(driver)
+      end
+
+      def name
+        'FuzzyUse'
+      end
+
+      #
+      # Returns the hash of commands supported by this dispatcher.
+      #
+      def commands
+        {
+          'fzuse' => 'A fuzzy use command added by the FuzzyUse plugin'
+        }
+      end
+
+      def pipe_server(socket_path)
+        server = UNIXServer.new(socket_path)
+        File.chmod(0600, socket_path)
+        loop do
+          client = server.accept
+          begin
+            unless (input_string = client.gets&.chomp).blank?
+              if (mod = framework.modules.create(input_string))
+                client.puts(Serializer::ReadableText.dump_module(mod))
+              end
+            end
+          rescue StandardError
+          end
+          client.close
+        end
+      rescue EOFError
+      ensure
+        server.close if server
+        File.delete(socket_path) if File.exist?(socket_path)
+      end
+
+      #
+      # This method handles the fzuse command.
+      #
+      def cmd_fzuse(*args)
```

Comment:
Just a question; Why is it called `fzuse` instead of `fzfuse` 👀

Reply:
I'm guessing the rationale is the `fzf` is considered an implementation detail, and in the future `fzuse` could technically be powered by anything? 🤔

Reply:
I went with `fzuse` because it's fuzzy-user. `fzf` is named as it is because it's fuzzy-finder.